backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -217,8 +230,11 @@ public async Task<AvailabilitySlotDto> GetAvailabilitySlotById(long id)
 
                 _context.Update(availabilitySlot);
             }
-            
-            _mapper.Map(updateAvailabilityDto.Schedule, availabilitySlot.Schedule);
+
+            if (!updateAvailabilityDto.Schedule.WithTeamMembers)
+            {
+                _mapper.Map(updateAvailabilityDto.Schedule, availabilitySlot.SlotMembers.First().Schedule);
+            }
 
             availabilitySlot.LocationType = updateAvailabilityDto.GeneralDetails!.LocationType;
```

Comment:
Why ! is used here?

Reply:
If you mean line 234, cause we use SlotMembers.First() only when we know there is only one slot member aka author, if you mean 239, it's not my code, but it's used cause someone made general details nullable in the dto

Reply:
I meant line 239